File: app/src/main/java/org/ole/planet/myplanet/utilities/NotificationUtil.java
package org.ole.planet.myplanet.utilities;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Context;
import androidx.core.app.NotificationCompat;
import androidx.core.app.NotificationManagerCompat;


public class NotificationUtil {


    public static void create(Context context, int smallIcon, String contentTitle, String contentText) {
        NotificationManager manager =
                (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);

        NotificationCompat.Builder a = new NotificationCompat.Builder(context, "11");

        setChannel(manager);

        Notification noti = a
                .setContentTitle(contentTitle)
                .setContentText(contentText)
                .setSmallIcon(smallIcon)
                .setProgress(0, 0, true)
                .setAutoCancel(true).build();
        manager.notify(111, noti);

    }

    public static void cancel(Context context, int id) {
        NotificationManagerCompat nm = NotificationManagerCompat.from(context);
        nm.cancel(id);
    }

    public static void cancellAll(Context context) {
        NotificationManagerCompat nm = NotificationManagerCompat.from(context);
        nm.cancelAll();
    }

    public static void setChannel(NotificationManager notificationManager) {
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
            int importance = NotificationManager.IMPORTANCE_LOW;
            NotificationChannel notificationChannel = new NotificationChannel("11", "ole", importance);
            notificationManager.createNotificationChannel(notificationChannel);
        }
    }
}

File: app/src/main/java/org/ole/planet/myplanet/utilities/NotificationUtil.kt
package org.ole.planet.myplanet.utilities

import android.app.NotificationChannel
import android.app.NotificationManager
import android.content.Context
import android.os.Build
import androidx.core.app.NotificationCompat
import androidx.core.app.NotificationManagerCompat

object NotificationUtil {
    private const val CHANNEL_ID = "ole_sync_channel"
    private const val NOTIFICATION_ID = 111

    @JvmStatic
    @Synchronized
    fun create(context: Context, smallIcon: Int, contentTitle: String?, contentText: String?) {
        val manager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        setChannel(manager)

        val notification = NotificationCompat.Builder(context, CHANNEL_ID)
            .setContentTitle(contentTitle)
            .setContentText(contentText)
            .setSmallIcon(smallIcon)
            .setProgress(0, 0, false)  // Remove indeterminate progress
            .setOnlyAlertOnce(true)
            .setPriority(NotificationCompat.PRIORITY_LOW)
            .setAutoCancel(true)
            .build()

        manager.notify(NOTIFICATION_ID, notification)
    }

    @JvmStatic
    @Synchronized
    fun cancel(context: Context, id: Int = NOTIFICATION_ID) {
        val nm = NotificationManagerCompat.from(context)
        nm.cancel(id)
    }

    @JvmStatic
    fun cancelAll(context: Context) {
        val nm = NotificationManagerCompat.from(context)
        nm.cancelAll()
    }

    @JvmStatic
    fun setChannel(notificationManager: NotificationManager?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val importance = NotificationManager.IMPORTANCE_LOW
            val notificationChannel = NotificationChannel(
                CHANNEL_ID,
                "OLE Sync",
                importance
            )
            notificationChannel.description = "Notifications for synchronization process"
            notificationChannel.setSound(null, null)  // Disable sound
            notificationChannel.enableVibration(false)  // Disable vibration
            notificationManager?.createNotificationChannel(notificationChannel)
        }
    }
}